### PR TITLE
Fix compatibility with -tfConfigExists=false

### PR DESCRIPTION
**Pull request by JackFlukinger**
Hey @im2nguyen , this fixes `-tfConfigExists false` with #76 .

### CODE REVIEW

**Comment by im2nguyen** [comment]
Hey @JackFlukinger, this works when `-tfConfigExists=false`, but errors when it's set to true (or unset)

```
$ ./rover -workingDir example/random-test/ -tfConfigExists=true
2022/01/11 19:47:56 Starting Rover...
2022/01/11 19:47:56 Initializing Terraform...
2022/01/11 19:47:57 Generating plan...
2022/01/11 19:47:58 Parsing configuration...
2022/01/11 19:47:58 Generating resource overview...
2022/01/11 19:47:58 Generating resource map...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x16606af]

goroutine 1 [running]:
main.(*rover).GenerateModuleMap(0xc00018f320, 0xc000406e10, {0xc000124978, 0x11})
        /Users/dos/go/src/github.com/JackFlukinger/rover/map.go:212 +0x1acf
main.(*rover).GenerateModuleMap(0xc00018f320, 0xc0003f7b88, {0x0, 0x0})
        /Users/dos/go/src/github.com/JackFlukinger/rover/map.go:240 +0xefc
main.(*rover).GenerateMap(0xc00018f320)
        /Users/dos/go/src/github.com/JackFlukinger/rover/map.go:319 +0x1bc
main.(*rover).generateAssets(0xc00018f320)
        /Users/dos/go/src/github.com/JackFlukinger/rover/main.go:207 +0x176
main.main()
        /Users/dos/go/src/github.com/JackFlukinger/rover/main.go:142 +0xd19
```

**Comment by im2nguyen** [comment]
This also changes some of the visualizations 🤔 

This the visualization generated by the binary produced by #76 (which is consistent with prior versions of rover)

image

This is the visualization generated by this PR. Notice that under the random cat module, the `random_name` and `max_length` variables are missing. Also, the filename (`main.tf`) under the random cat module is also missing

image

**Comment by JackFlukinger** [comment]
Whoops, will fix.

File: map.go [hunk 1]
```diff
@@ -195,9 +219,7 @@ func (r *rover) GenerateModuleMap(parent *Resource, parentModule string) {
```
Comment (im2nguyen) [inline map.go:219]: if you change lines 212 to the following it resolves the error I mention in https://github.com/im2nguyen/rover/pull/79#issuecomment-1010603958

```
else if rs.Type == ResourceTypeResource {
	r := configs[parentConfig].Module.ManagedResources[ind]

	if r != nil {
		fname = filepath.Base(r.Pos.Filename)
		re.Line = &r.Pos.Line
	}
}
```

**Comment by im2nguyen** [comment]
Based on going through the `rso.json`, it seems like all the items are there. I think it's tripping up because it's not parsing the file name even if the configuration exists.

This is the command I'm using to test the command without rebuilding it every time

```
go run . -workingDir example/random-test
```

**Comment by JackFlukinger** [comment]
Think this is actually related to how it handles -workingDir flag in map generation. I'll commit a fix in a bit.

**Comment by JackFlukinger** [comment]
Should be fixed @im2nguyen , let me know if it looks good on your end!

**Comment by im2nguyen** [comment]
that worked! thanks!! 😄

**Pull request merged**